index.js

```diff
@@ -11,44 +12,119 @@ function constructRequest(har, userAgent = false) {
     method: request.method,
   };
 
-  if (request.headers.length) {
-    options.headers = request.headers.map(header => headers.append(header.name, header.value));
+  if ('headers' in request && request.headers.length) {
+    request.headers.forEach(header => headers.append(header.name, header.value));
+  }
+
+  if ('cookies' in request && request.cookies.length) {
+    // As the browser fetch API can't set custom cookies for requests, they instead need to be defined on the document
+    // and passed into the request via `credentials: include`. Since this is a browser-specific quirk, that should only
+    // happen in browsers!
+    if (typeof window !== 'undefined' && typeof document !== 'undefined') {
+      request.cookies.forEach(cookie => {
+        document.cookie = `${encodeURIComponent(cookie.name)}=${encodeURIComponent(cookie.value)}`;
```

Comment:
Wouldn't this overwrite the cookie if there are multiple cookies? Maybe I'm not understanding the intent here

Reply:
NVM: https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie

Reply:
Really interesting API here haha

Reply:
Nope, it's additive:

```
> document.cookie;
"_ga=GA1.2.2144919228.1560793249; _octo=GH1.1.1087835889.1560793249; tz=America%2FLos_Angeles; _gat=1"
> document.cookie = 'tk="test"';
"tk="test""
> document.cookie;
"tk="test"; _ga=GA1.2.2144919228.1560793249; _octo=GH1.1.1087835889.1560793249; tz=America%2FLos_Angeles; _gat=1"
```